cmd/apiProviderDownload_test.go

```diff
@@ -0,0 +1,76 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type apiProviderDownloadMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func TestRunApiProviderDownload(t *testing.T) {
```

Comment:
As a reviewer, I would like to get orderly, easy-to-understand tests, covering the all good and bad cases which on one hand a consumer could trigger, on the other hand the code expects.

Reconsider:
- do the tests cover the main cases? (like "downloadPath does no exists", good or bad?, "provider does no exists" how does the code behave?, ...)
- can you easily distinguish the cases (move all standard setups into functions)

Reply:
I have re-structured the tests accordingly. Kindly review.

Reply:
Could you refer to the commits where this was addressed? I'm asking because there are two tests and when I would prefer many functionality specific tests that show us (the reviewers) how the code works.

Reply:
Fixed review comments on documentation and test structuring

commit d22548ef12cb0c8a86818c82815a6dbeb125be12